plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-product-data.php

```diff
@@ -293,8 +295,13 @@ public static function prepare_attributes( $data = false ) {
 	private static function prepare_set_attributes( $all_attributes, $key_prefix = 'attribute_', $index = null ) {
 		$attributes = array();
 
-		if ( $all_attributes ) {
+		if ( is_array( $all_attributes ) ) {
```

Comment:
This method is used when saving meta box data -- both overall and the variation-specific one. So [this comment](prepare_set_attributes) applies here as well, and the same issues (no proper way to gracefully disrupt the flow) apply.

Reply:
let's revert this too based on your description